### PR TITLE
HtmlMesh Extension

### PR DESCRIPTION
**Pull request by ericwood73**
The HtmlMesh is a [BabylonJS](https://babylonjs.com) mesh that displays a webpage (or DOM element) 
in the scene, meaning that it can occlude other meshes and be occluded by other meshs. 

[Online Demo](https://codesandbox.io/p/sandbox/babylon-html-mesh-demo-862gh5)

[BabylonJS Playground](https://www.babylonjs-playground.com/full.html#Y2LIXI#14).  Note the currently only works in fullscreen.  There is an issue with getting the wrong canvas size initially.

The following uses cases are supported by the HtmlMesh
* Add instructional content/video to a 3D scene.
* Allow users in a 3D scene to fill out a form or respond to a poll.
* Display an in-app browser so a user can take some action, e.g. sign up for an account, purchase something, join an email list without leaving the experience.
* Create a rich UI that can leverage all of the capabilities of CSS and can be driven through a CMS

Note that the use of HtmlMesh requires that the experience be accessed through a browser so in it's
current form, this will not work in native apps or in XR.  If constructed outside of a browser context, the
HtmlMesh instances will not have any geometry and will be disabled.  In the future, it might make sense to 
have them appear with placeholder content of perhaps even an option to view the raw HTML in a popup UI.

### CODE REVIEW

File: .gitignore
Comment (sebavan) [inline .gitignore:3]: Could those updates be moved in your specific gitIgnore ?

File: HtmlMesh/example.js
Comment (sebavan) [inline HtmlMesh/example.js:152]: Should this part be a separated example ?

**Review by sebavan — COMMENTED** [review]
Thanks a lot for the project !!!!

Only minor comments for now.

Would be great to remove empty functions comments as despite helping linters they are not useful to the contributors. I d suggest to either fill them in or remove them.

I ll go depper after those.

File: HtmlMesh/example.js
Comment (ericwood73) [reply HtmlMesh/example.js:152]: Yes, I can remove.  I started to create an example of how to do pointer masking but then realized you'd have to CSS transform the mask and so I left it for something to do maybe do later.

File: .gitignore
Comment (ericwood73) [reply .gitignore:3]: Will do

**Comment by ericwood73** [comment]
@sebavan Think I addressed all of your comments.

**Comment by sebavan** [comment]
starting a deeper pass :-)

File: .gitignore
Comment (sebavan) [inline .gitignore:3]: ohhhh my bad, you should revert this one, did not notice yesterday it was already in... so sorry

File: HtmlMesh/README-npm.md
Comment (sebavan) [inline HtmlMesh/README-npm.md:183]: You should remove left over lines. The rest is amazing !!!

File: HtmlMesh/README.md
Comment (sebavan) [inline HtmlMesh/README.md:183]: Same comment on this one.

File: HtmlMesh/README-npm.md
Comment (sebavan) [inline HtmlMesh/README-npm.md:1]: Why not keeping only readme.md if the content is the same ?

**Review by sebavan — APPROVED** [review]
You should re-indent the TS files in src with prettier as the current indentation looks a bit off. Aside of this great job !!!!

I am approving so you can merge once the little last "nits" have been addressed.

File: HtmlMesh/README-npm.md
Comment (ericwood73) [reply HtmlMesh/README-npm.md:1]: The npm readme is slightly different.  E.g. it does not have instructions for running locally.

Comment (ericwood73) [reply HtmlMesh/README-npm.md:1]: I see.  They should be different but are the same in this commit.

**Comment by ericwood73** [comment]
@sebavan I don't see an option to merge.  Maybe I don't have merge rights?

**Comment by sebavan** [comment]
All done !!! thanks for the contribution !!!!

**Pull request merged**
